### DISCUSSION TITLE
Why symfony/clock only on v6.2

### DISCUSSION THREAD

**Discussion by bzajacc** [symfony/symfony#47560] [Q&A] [answered]
Hi!
Why this component https://github.com/symfony/clock is only available on 6.2 version? 

Can we add this to older ones, at least 6.1? :)

**Comment by MatTheCat** [comment, marked as answer]
Patch versions cannot include new features. Symfony 6.1 was released on May 2022 while symfony/clock started Jul 28, 2022 so it had to be on the 6.2 branch.

That being said nothing prevents you to require it, whatever the version of your other Symfony components (as long as they don’t conflict).

**Comment by bzajacc** [comment]
Yes I know abut patch flow, but its still new component, that not affect others, and I'm just confused why this is not on every tag/version branch. Why it is not on 5.4, 6.0, 6.1 etc. branch as well on start?

I can't include because my project is restrict to 6.1.*

**Reply by MatTheCat** [reply]
Please use “write a reply” to reply instead of “suggest an answer”.

> Why it is not on 5.4, 6.0, 6.1 etc. branch as well on start?

Because it started on 6.2. New features cannot be backported since patch versions cannot include them.

> I can't include because my project is restrict to 6.1.*

Well don’t restrict symfony/clock then (I don’t even know how you could restrict a package you did not require :thinking: ).

**Reply by bzajacc** [reply]
> Please use “write a reply” to reply instead of “suggest an answer”.

Oh sorry, "miss click" :)

> (I don’t even know how you could restrict a package you did not require 🤔 ).

In composer in extra, something like this 😄 

```json
{
  "extra": {
      "symfony": {
          "allow-contrib": false,
          "require": "6.1.*"
      }
    }
}
```

**Reply by MatTheCat** [reply]
AFAIK `extra.symfony.require` should not prevent you to run e.g.

```shell
composer require "symfony/clock 6.2.*-dev"
```

**Reply by bzajacc** [reply]
Damn, earlier I try this and not working... 🥲  But thanks this is good enough for me! 😃

**Reply by MatTheCat** [reply]
You're welcome ^^

I guess you can mark this discussion as answered.
